var/spack/repos/builtin/packages/care/package.py

```diff
@@ -0,0 +1,129 @@
+        spec = self.spec
+
+        options = []
+        options.append('-DBLT_SOURCE_DIR={0}'.format(spec['blt'].prefix))
+
+        if '+cuda' in spec:
+            options.extend([
+                '-DENABLE_CUDA=ON',
+                '-DCUDA_TOOLKIT_ROOT_DIR=' + spec['cuda'].prefix])
+
+            if not spec.satisfies('cuda_arch=none'):
+                cuda_arch = spec.variants['cuda_arch'].value
+                options.append('-DCUDA_ARCH=sm_{0}'.format(cuda_arch[0]))
+                flag = '-arch sm_{0}'.format(cuda_arch[0])
+                options.append('-DCMAKE_CUDA_FLAGS:STRING={0}'.format(flag))
+        else:
+            options.append('-DENABLE_CUDA=OFF')
+
+        if '+hip' in spec:
+            arch = self.spec.variants['amdgpu_target'].value
+            options.extend([
+                '-DENABLE_HIP=ON',
+                '-DHIP_ROOT_DIR={0}'.format(spec['hip'].prefix),
+                '-DHIP_HIPCC_FLAGS=--amdgpu-target={0}'.format(arch)])
+        else:
+            options.append('-DENABLE_HIP=OFF')
+
+        options.append('-DCARE_ENABLE_IMPLICIT_CONVERSIONS={0}'.format(
+            'ON' if '+implicit_conversions' in spec else 'OFF'))
+
+        options.append('-DCAMP_DIR:PATH='
+                       + spec['camp'].prefix.share.camp.cmake)
+        options.append('-DUMPIRE_DIR:PATH='
+                       + spec['umpire'].prefix.share.umpire.cmake)
+        options.append('-DRAJA_DIR:PATH='
+                       + spec['raja'].prefix.share.raja.cmake)
+        options.append('-DCHAI_DIR:PATH='
+                       + spec['chai'].prefix.share.chai.cmake)
+
+        options.append('-DCARE_ENABLE_TESTS={0}'.format(
+            'ON' if self.run_tests else 'OFF'))
+
+        # There are both CARE_ENABLE_* and ENABLE_* variables in here because
+        # one controls the BLT infrastructure and the other controls the CARE
+        # infrastructure. The goal is to just be able to use the CARE_ENABLE_*
+        # variables, but CARE isn't set up correctly for that yet.
+        options.append('-DENABLE_BENCHMARKS={0}'.format(
+            'ON' if '+benchmarks' in spec else 'OFF'))
+        options.append('-DCARE_ENABLE_BENCHMARKS={0}'.format(
+            'ON' if '+benchmarks' in spec else 'OFF'))
+
+        options.append('-DENABLE_EXAMPLES={0}'.format(
+            'ON' if '+examples' in spec else 'OFF'))
+        options.append('-DCARE_ENABLE_EXAMPLES={0}'.format(
+            'ON' if '+examples' in spec else 'OFF'))
+
+        options.append('-DENABLE_DOCS={0}'.format(
+            'ON' if '+docs' in spec else 'OFF'))
+        options.append('-DCARE_ENABLE_DOCS={0}'.format(
+            'ON' if '+docs' in spec else 'OFF'))
+
+        # give clear error for conflict between self.run_tests and
+        # benchmarks variant.
+        if not self.run_tests and '+benchmarks' in spec:
+            raise InstallError(
+                'ENABLE_BENCHMARKS requires ENABLE_TESTS to be ON'
+            )
```

Comment:
I think this should be reworked so that `+benchmarks` is authoritative for building tests. That will require changing one spot above as well.

`self.run_tests` just controls whether to run `make check` as part of the build. If I understand correctly, `make check` can still run even if the benchmarks aren't built.

Reply:
I added a separate +tests variant and conflicts with benchmarks.

For now, turning on +tests fails to build because it cannot find gtest. I'll investigate that in a follow up pull request and then turn on tests by default as appropriate.

Reply:
"Might need a `test` type dependency on `gtest`

Can you add `conflicts('+test')` until that is resolved, so we know not to expect `care+test` to install?

Reply:
Added that here and in a few other projects with the same issue